backend/src/main/java/com/google/sps/servlets/SearchMatchServlet.java

```diff
@@ -44,38 +48,41 @@
   private static final String JSON_SECOND_PARTICIPANT_USERNAME = "secondParticipantUsername";
   private static final String JSON_DURATION = "duration";
 
+  // Create GmailFactory and use it to instantiate Email Notifier
+  private final GmailFactory gmFactory = new GmailFactory();
+  // Get DatastoreService and instiate Match and Participant Datastores
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private final MatchDatastore matchDatastore = new MatchDatastore(datastore);
+  private final ParticipantDatastore participantDatastore = new ParticipantDatastore(datastore);
+  // Get participant username
+  UserService userService = UserServiceFactory.getUserService();
+  private EmailNotifier emailNotifier;
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
     System.out.println("Request received");
 
-    // Get participant username
-    UserService userService = UserServiceFactory.getUserService();
-    String email = userService.getCurrentUser().getEmail();
-    if (email == null) {
-      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Invalid email.");
-      return;
+    try {
+      emailNotifier = ensureEmailNotifier();
+    } catch (GeneralSecurityException e) {
```

Comment:
Is this used outside of sendExpiredResponse? If not can this block move into that method?

Reply:
It is used for both send expired and sendMatch, so I'll move the logic to those two places !